Add info about Workflow API token in REST API (#2295)
astronomer/docs: astro/airflow-api.md

## Prerequisites

- A Deployment on Astro.
- A [Workspace API token](workspace-api-tokens.md) or an [Organization API token](organization-api-tokens.md).
- [cURL](https://curl.se/) or, if using Python, the [Requests library](https://docs.python-requests.org/en/latest/index.html).
- The [Astro CLI](cli/overview.md).

## Step 1: Retrieve your access token

<Tabs groupId="step-1-retrieve-your-access-token">

<TabItem value="workspace" label="Workspace token (Recommended)">

> You can follow the steps in [Create a Workspace API token](workspace-api-tokens.md#create-a-workspace-api-token) to create your token. You will need to save the token on creation in order to re-use it later.

Reviewer: ```suggestion
Follow the steps in [Create a Workspace API token](workspace-api-tokens.md#create-a-workspace-api-token) to create your token. Make sure to save the token on creation in order to use it later in this setup.
```

</TabItem>

<TabItem value="organization" label="Organization token">

> You can follow the steps in [Create a Orgaization API token](organization-api-tokens.md#create-an-organization-api-token) to create your token. You will need to save the token on creation in order to re-use it later.

Reviewer: ```suggestion
Follow the steps in [Create a Orgaization API token](organization-api-tokens.md#create-an-organization-api-token) to create your token. Make sure to save the token on creation in order to use it later in this setup.
```

## Prerequisites

- A Deployment on Astro.
- A [Workspace API token](workspace-api-tokens.md) or an [Organization API token](organization-api-tokens.md).
- [cURL](https://curl.se/) or, if using Python, the [Requests library](https://docs.python-requests.org/en/latest/index.html).
- The [Astro CLI](cli/overview.md).

## Step 1: Retrieve your access token

<Tabs groupId="step-1-retrieve-your-access-token">

<TabItem value="workspace" label="Workspace token (Recommended)">

> Workspace API tokens allow users to programmatically perform Workspace actions as well as Airflow API access. 

Reviewer: ```suggestion
Workspace API tokens allow users to programmatically perform Workspace actions, including accessing the Airflow API for Deployments in the Workspace.
```

> Keep in mind the following when creating Workspace API tokens for Airflow REST API access:

Reviewer: ```suggestion
Keep in mind the following when using Workspace API tokens to access the Airflow REST API:
```

Reviewer: This seem more like general benefits of the Workspace API tokens than things to "Keep in mind" when using them. I think we can remove this bulleted list.

Author: I think as a reference it is good, what do you think would be an alternate heading ?

Reviewer: I don't think we should really have it in this doc at all, because this isn't the info a user is interested in when they are trying to access the Airflow API. We should keep it short and straight to the point: 

"In most use cases, Astronomer recommends using Workspace API tokens to access the Airflow REST API. See [Workspace API tokens](workspace-api-tokens.md#create-a-workspace-api-token) to create a token.

"After you create a token, add it to the Authorization HTTP header in the following format:"

We can include the info about the benefits of the Workspace API tokens on the token documentation itself. Adding the list of benefits here is another barrier to the user learning how to access the Airflow API.

- You can create these tokens using the Cloud UI only and can also set the expiration period for it. 
- You can assign [Workspace roles](user-permissions.md#workspace-roles) to the API token during creation based on the actions you want to perform.
- Workspace API tokens are an easy way to control access to all your Deployments that you want to manage with same set of permissions.

> Refer to [Workspace API tokens](workspace-api-tokens.md#create-a-workspace-api-token) to create a token.

Reviewer: Do we want to include a sample cURL request that uses the API token? What about how to make the token accessible to the system calling the Airflow API?

</TabItem>

<TabItem value="organization" label="Organization token">

Organization API tokens allow users to programmatically perform Organization operations in Astro as well as Airflow API access. 

> Keep in mind the following when creating Organization API tokens for Airflow REST API access:

Author: > Looking good in general @manmeetkaur ! Marking changes because I think we need to add some more general code examples for how to use tokens to call the API. How do users make the token accessible to the system calling the Airflow API? We should list those options here.

@jwitz I think we already discussed this during another PR review that we need in general a page that explains about generating tokens and using in API, that will remove a lot of repetitive code/info wherever we are referring to creating tokens. (I have forgotten the PR # but I remember David and you being part of it - the problem I face with long -pending PRs 🤣 )

> Keep in mind the following when creating Organization API tokens for Airflow REST API access:

Reviewer: Ugh sorry I lost that thread.. I think we just need some basic info here about where to add the token to a request. We don't have to get crazy with _all_ the options or add too many code samples, but just like one simple instruction for how to use the token _after_ you've generated it.

- You can create these tokens using the Cloud UI only and can also set the expiration period for it. 
- Organization API token with `Organization Member` role will not be able to access any Workspaces or Deployments. You will need to give permission to it to a specific Workspace. See Give Workspace permissions to an Organization token for more details.
- You can assign [Organization roles](user-permissions.md#user-permissions#organization-roles) to the API token during creation based on the actions you want to perform.
- Organization API tokens are an easy way to control access to all your Workspaces and Deployments, so you might want to use and share them with caution.

> Refer to [Organization API tokens](organization-api-tokens#create-an-organization-api-token) to create a token.

Reviewer: Do we want to include a sample cURL request that uses the API token?

Author: I think last time when we talked about this, we decided to have a re-look at this and other API token methods that can be used as a reference examples.

Reviewer: Could we do something generalized? Like just saying "After you create a token, add it to the `Authorization` HTTP header in the following format:

```sh
curl -X GET <your-deployment-url>/api/v1/<api-endpoint>
curl -X GET https://<your-deployment-url>/api/v1/<api-endpoint> \
   -H 'Cache-Control: no-cache' \
   -H 'Authorization: Bearer <your-access-token>'
```

</TabItem>

<TabItem value="deployment" label="Deployment API key">

> :::info

Reviewer: ```suggestion
:::caution
```

> Deployment API keys will soon be phased out in favor of Deployment-level API tokens. Astronomer recommends using Workspace tokens to reach the Airflow API until Deployment API tokens are released. 

Reviewer: ```suggestion
Deployment API keys will soon be phased out in favor of Deployment-level API tokens. Astronomer recommends only using Workspace API tokens until Deployment API tokens are released. 
```

</TabItem>
</Tabs>

## Step 2: Retrieve the Deployment URL

Run the following command to retrieve a Deployment URL:

```bash
astro deployment inspect -n <deployment-name> -k metadata.webserver_url
```

The Deployment URL is used to access your Astro Deployment's Airflow UI. It includes the name of your Organization and first 7 letters of your Deployment ID. 

For example, the Deployment URL for an Organization named `mycompany` with the Deployment ID `dhbhijp0vt68400dj40tmc8virf` is `mycompany.astronomer.run/dhbhijp0`.

> ## Step 2: Make an Airflow API request

Reviewer: ```suggestion
## Step 3: Make an Airflow API request
```

</TabItem>
</Tabs>

## Step 2: Retrieve the Deployment URL

Run the following command to retrieve a Deployment URL:

```bash

> astro deployment inspect -n my-deployment-name -k metadata.webserver_url

Author: the reason I changed it because `>` was also used in this page as a command. It was confusing to see `> <` side by side. I can put back the placeholder formatting.
